test/org/ojalgo/scalar/RationalNumberTest.java

```diff
@@ -0,0 +1,66 @@
+package org.ojalgo.scalar;
+
+import org.ojalgo.FunctionalityTest;
+import org.ojalgo.TestUtils;
+import org.ojalgo.constant.PrimitiveMath;
+
+import java.math.BigDecimal;
+
+import static java.lang.Double.longBitsToDouble;
+
+public class RationalNumberTest extends FunctionalityTest {
+
+    private final double myDiff = PrimitiveMath.MACHINE_EPSILON;
+
+    public void testValueOf() {
+
+        double test_values[] = {
+                //                 seeeeeeeeeemmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmm
+                longBitsToDouble(0b011111110111001100110011001100110011001100110011001100110011010L), // 0.1
+                longBitsToDouble(0b011111101011111111111111111111111111111111111111111111111111111L), // * 2^{-62}
+                longBitsToDouble(0b011111101001111111111111111111111111111111111111111111111111111L), // * 2^{-63}
```

Comment:
So this is the `MAX_BITS` case; it will fail if we change the shift logic back. At the same time, that seems to be "fixing" the SV decomposition tests.